spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/hystrix/security/SecurityContextRegistratorCommandHook.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.hystrix.security;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import com.netflix.hystrix.HystrixInvokable;
+import com.netflix.hystrix.strategy.concurrency.HystrixRequestContext;
+import com.netflix.hystrix.strategy.executionhook.HystrixCommandExecutionHook;
+
+/**
+ * @author Daniel Lavoie
+ */
+public class SecurityContextRegistratorCommandHook extends HystrixCommandExecutionHook {
+	private static final Logger LOGGER = LoggerFactory
+			.getLogger(SecurityContextRegistratorCommandHook.class);
+
+	private CustomCommandExecutionHook customCommandHook;
+
+	public SecurityContextRegistratorCommandHook(CustomCommandExecutionHook customCommandHook) {
+		this.customCommandHook = customCommandHook;
+	}
+
+	@Override
+	public <T> void onExecutionStart(HystrixInvokable<T> commandInstance) {
+		if (LOGGER.isTraceEnabled())
+			LOGGER.trace("Executing security registration command hook.");
+
+		if (!HystrixRequestContext.isCurrentThreadInitialized()) {
+			if (LOGGER.isTraceEnabled())
+				LOGGER.trace("Initializing Hystrix Request Context");
+
+			HystrixRequestContext.initializeContext();
+		}
+
+		SecurityContext context = SecurityContextHystrixRequestVariable.getInstance()
+				.get();
+		if (context != null) {
+			if (LOGGER.isTraceEnabled())
+				LOGGER.trace("Injecting security context.");
+
+			SecurityContextHolder.setContext(context);
+		}
+
+		if (customCommandHook != null)
+			customCommandHook.onExecutionStart(commandInstance);
+	}
+}
```

Comment:
I just started using Hystrix, so this question may be a result of my ignorance. I'm curious why you implemented a command execution hook instead of a concurrency strategy that overrides the [wrapCallable](https://github.com/Netflix/Hystrix/wiki/Plugins#concurrencystrategy) method. For example:
```java
import com.netflix.hystrix.strategy.concurrency.HystrixConcurrencyStrategy;
import java.util.concurrent.Callable;
import org.springframework.security.concurrent.DelegatingSecurityContextCallable;

public class SecurityContextConcurrencyStrategy extends HystrixConcurrencyStrategy {
    @Override
    public <T> Callable<T> wrapCallable(Callable<T> callable) {
        return super.wrapCallable(new DelegatingSecurityContextCallable(callable));
    }
}
```
After the executing the callable, `DelegatingSecurityContextCallable` clears the security context from the thread.